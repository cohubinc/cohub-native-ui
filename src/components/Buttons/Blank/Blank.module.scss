.ButtonBlank {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  cursor: pointer;
  outline: none;
  text-align: inherit;
  border-radius: 0;
}
