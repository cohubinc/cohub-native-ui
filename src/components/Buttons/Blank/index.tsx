import React, {
  DetailedHTMLProps,
  ButtonHTMLAttributes,
  LegacyRef,
} from 'react';
import pick from 'lodash/pick';

import Typography from '../../Typography';
// import styles from './Blank.module.scss';

// When you want to wrap another element in a button but don't want any button styles from the browser.
// If it's clickable it should probably be a button or a link
export type Props = {
  nativeElRef?: LegacyRef<HTMLButtonElement>;
} & DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;
const Blank = ({
  className = '',
  children,
  style,
  nativeElRef,
  ...rest
}: Props) => (
  <button
    style={style}
    // className={`${styles.ButtonBlank} ${className}`}
    ref={nativeElRef}
    {...rest}
  >
    <Typography style={pick(style, 'color', 'fontSize')}>{children}</Typography>
  </button>
);

export default Blank;
