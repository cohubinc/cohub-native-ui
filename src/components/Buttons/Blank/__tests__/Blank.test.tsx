import React from "react";
import { mount } from "enzyme";

import Blank from "../index";

describe("Blank", () => {
  it("renders without crashing", () => {
    const wrapper = mount(<Blank>push me</Blank>);
    expect(wrapper).toMatchSnapshot();
  });
});
