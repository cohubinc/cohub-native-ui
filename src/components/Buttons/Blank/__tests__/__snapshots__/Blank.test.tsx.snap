// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank renders without crashing 1`] = `
<Blank>
  <button
    className="ButtonBlank "
  >
    <Typography
      style={Object {}}
    >
      <Component
        style={Object {}}
      >
        <span
          className=""
          data-qa="text"
          style={
            Object {
              "color": "hsl(0, 2%, 22%)",
              "fontFamily": "Inter",
              "fontSize": "14px",
              "fontWeight": 400,
              "textAlign": "initial",
              "textTransform": "initial",
            }
          }
        >
          push me
        </span>
      </Component>
    </Typography>
  </button>
</Blank>
`;
